src/main/java/frc/robot/vision/PhotonVisionIOReal.java

```diff
@@ -0,0 +1,89 @@
+package frc.robot.vision;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Transform3d;
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonPoseEstimator;
+
+public class PhotonVisionIOReal implements VisionIO {
+
+    private final PhotonCamera camera;
+    private final PhotonPoseEstimator estimator;
+    private final Transform3d robotToCamera;
+    private Result result;
+
+    public PhotonVisionIOReal(
+            PhotonCamera camera, Transform3d robotToCamera, AprilTagFieldLayout field) {
+        this.camera = camera;
+        this.robotToCamera = robotToCamera;
+        camera.setPipelineIndex(0);
+        try {
+            estimator =
+                    new PhotonPoseEstimator(
+                            field,
+                            PhotonPoseEstimator.PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
+                            camera,
+                            robotToCamera);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void setPipeLine(int pipeLineIndex) {
+        camera.setPipelineIndex(pipeLineIndex);
+    }
+
+    @Override
+    public void updateInputs(VisionInputs inputs) {
+        var latestResult = camera.getLatestResult();
```

Comment:
[`getMultiTagResult()`](https://photonvision.github.io/photonvision/built-docs/javadoc/org/photonvision/targeting/PhotonPipelineResult.html#getMultiTagResult()) should be used instead, because we use the `MULTI_TAG_PNP_ON_COPROCESSOR` mode. Note that the pose returned from this method is relative to the [field origin](https://docs.wpilib.org/en/stable/docs/software/advanced-controls/geometry/coordinate-systems.html), not to the robot.

Reply:
Do we have to? The whole idea was to also get some data about the current target we're seeing. Is there some way to do that, and keep the multi tag pnp on the coprocessor?

Reply:
If I understand correctly, the method you use now doesn't return the fused MultiTag pose. Also, why do you need to see the tag itself?

Reply:
We just want the ability to understand what the camera sees, it helps sometimes with debugging.